Exploration Queries/SecurityEvent/ParentProcessesOnHost.txt

```diff
@@ -1,35 +1,32 @@
-// Name: Parent Processes running on Host
-// Description: Any 4688 event that contains Parent Process running on a given host during the time of or recently after compromise and in the range of +-10m
-// Entity: #Host
-// Input: Computername, Hostname, HostId
-// Output: Processes
-//
-// QueryPeriod: -1h and +30m default, change as needed, not all event 4688 have Parent Process available
-//
-// Data Source: #SecurityEvent
-//
+// Id: 07da3cc8-c8ad-4710-a44e-334cdcb7882b
+// DisplayName: Parent Processes running on Host
+// Description: Any 4688 event that contains Parent Process running on a given host during the time of or recently after compromise (set time range to +-10m when running the query)
+// InputEntityType: Host
+// InputFields: [HostName]
+// OutputEntityTypes: [Process]
+// QueryPeriodBefore: 10m
+// QueryPeriodAfter: 10m
+// DataSources: #SecurityEvent
 // Tactics: #Persistence, #Discovery, #LateralMovement, #Collection
-//
-let GetParentProcessesOnHost = (suspiciousEventTime:datetime, v_Host:string){
-let v_StartTime = suspiciousEventTime-10m;
-let v_EndTime = suspiciousEventTime+10m;
+
+let GetParentProcessesOnHost = (v_Host_HostName:string){
 SecurityEvent 
-| where TimeGenerated between (v_StartTime .. v_EndTime)
-| where toupper(Computer) contains v_Host or toupper(WorkstationName) contains v_Host
```

Comment:
lines of the format (toupper(), tolower(), etc...) are computationally heavier than other simple clauses. So it is good to try and push them as much further down the query as possible so they do not affect the performance of the query